src/rn.h

```diff
@@ -522,11 +522,12 @@ template <typename T>
 T RN<T>::get_order(T x)
 {
   if (x == 0 || x == 1) return 1;
-  std::vector<T> primes;
-  std::vector<T> exponent;
   T h = this->card_minus_one();
-  _factor_prime<T>(h, &primes, &exponent);
-  T order = do_step_get_order(x, h, &primes, &exponent);
+  if (!this->compute_factors_of_order_done)
+    _factor_prime<T>(h, this->primes, this->exponent);
+  std::vector<T> _primes(*primes);
+  std::vector<T> _exponent(*exponent);
+  T order = do_step_get_order(x, h, &_primes, &_exponent);
```

Comment:
@lamphamsy do we need exponent and order as T ? can't just be integers ? (maybe stupid question and we need it)

Reply:
Since order could be (q-1), it should be as T. For exponent, I think it could be integers :)